### ISSUE TITLE
[Report] Steam Deck Client UI won't work after update

### ISSUE THREAD

**Issue by ftruncale** [holoiso-eol/holoiso#564]
## Your Issue:
```
The Deck UI doesn't seem to work, but plasma does fine. Appears to be a vulkan issue?
```

## Your device's full specs
```
Steam Deck
https://www.steamdeck.com/en/tech/ 
```

## Your OS version (`cat /etc/os-release | grep VARIANT_ID | cut -d '"' -f 2`):
```
snapshot19700101_1011-beta2a
```

## (Only for SteamOS Session loading issues) Have you tried launching `gamescope-session` from terminal? If yes, provide a log.
```
╰─ gamescope-session                                                                                         ─╯
no Intel GPU, skipping color compression disabling
Claimed global gamescope stats session at "/run/user/1000/gamescope-stats"
wlserver: [backend/headless/backend.c:82] Creating headless backend
ATTENTION: default value of option vk_xwayland_wait_ready overridden by environment.
vulkan: failed to find physical device
Failed to initialize Vulkan
 
```

## Your installation type:
```
deckperience
```

## Have you updated your installation recently?
```
Yes, there was a friday steam deck update through the client ui.
```

**Comment by suboxide** [comment]
same error on my steamdeck

**Comment by suboxide** [comment]
ok found my solution, boot up till you get stuck (mine kept getting stuck loading ramdisk), press ctrl+alt+F3 to get into tty and login.
Do a "ls -l /var/cache/pacman/pkg/vulkan*" then look for the one updated today and roll it back with: 
sudo pacman -U /var/cache/pacman/pkg/vulkan-radeon-22.2.0_devel.1567 (tab, and sudo reboot)

**Comment by RustyStriker** [comment]
Seems @suboxide is correct and it was a faulty `vulkan-radeon` update, for  sake of reference i downgraded from
`vulkan-radeon-22.2.0_devel.156801.jupiter_22.3.2-2` to `vulkan-radeon-22.2.0_devel.156798.jupiter_22.3.1-1` and steam(actually without a reboot even) launched

**Comment by ftruncale** [comment]
@RustyStriker @suboxide thanks for the help! here I was trying to reinstall the `vulkan-radeon` package and getting no luck. downgrade works great!

**Comment by sunhotblaze** [comment]
[Downgrade package](https://steamdeck-packages.steamos.cloud/archlinux-mirror/jupiter-main/os/x86_64/vulkan-radeon-22.2.0_devel.156793.jupiter_22.3.0-1-x86_64.pkg.tar.zst)
Test available after installation !

**Comment by S1ngl3-x** [comment]
> 
Thank you it worked. Will this be broken again with another update?

**Comment by theVakhovskeIsTaken** [comment]
> > 
> Thank you it worked. Will this be broken again with another update?
> 

It is broken because it will be updated again

What's happening? Valve finally updated their Arch Linux base, and that includes LLVM/Glibc/GCC. The new MESA version requires LLVM version not lower than 14

And those changes are in *-main branches, which correspond to Beta/RC channel on Steam Deck. 

What you're gonna wait for, is updates for main image/update client to make it update properly from SteamOS base 3.3 to SteamOS base 3.3.2/3.4-next

Same thing applies for new installations, where Valve nuked yay-git as a package and all new installs simply fail.


I'm not really in the condition/mood to fix this, but i will try my best to push it ASAP

**Comment by nagybal** [comment]
Should this work alteady? Is there a workaround for this issue?

**Comment by calamityjoe87** [comment]
> Should this work alteady? Is there a workaround for this issue?

It wasn't for me. I built the latest ISO from Actions, did a fresh install and was greeted to a black screen (on Radeon 6700XT). I was able to fix it by dropping into TTY (ctrl+atl+F2) and starting the plasma session:

`startplasma-wayland`

I manually downloaded an older version of `vulkan-radeon` ([vulkan-radeon-22.2.0_devel.156798.jupiter_22.3.1-1-x86_64.pkg.tar.zst](https://steamdeck-packages.steamos.cloud/archlinux-mirror/jupiter-main/os/x86_64/vulkan-radeon-22.2.0_devel.156798.jupiter_22.3.1-1-x86_64.pkg.tar.zst)) and forced the install using

`sudo pacman -U vulkan-radeon-22.2.0_devel.156798.jupiter_22.3.1-1-x86_64.pkg.tar.zst`

After a reboot, I was able to go into DeckUI mode. I didn't get any errors from pacman when doing this.
